token.go

```diff
@@ -9,8 +10,22 @@ import (
 // the key for verification.  The function receives the parsed, but unverified
 // Token.  This allows you to use properties in the Header of the token (such as
 // `kid`) to identify which key to use.
+//
+// The returned interface{} may be a single key or a PublicKeyset containing
+// multiple keys.
 type Keyfunc func(*Token) (interface{}, error)
 
+// PublicKey represents a generic public key interface.
```

Comment:
I think this would be the `DecodingKey` since it's the public key or the symmetric key.  The signing key is the private key generally.

Reply:
> I think this would be the "validating/decoding key" since it's the public key or the symmetric key. The signing key is the private key generally. Maybe `DecodingKey` or `ValidationKey`?

Ah yes of course.

Reply:
How about the term "verify" (either `VerifyKey` or `VerificationKey`), since this is where the keys are used? 

For example (using the shorter of the two):

```go
// VerifyKey represents a public or secret key for verifying a token's signature.
type VerifyKey interface {
	crypto.PublicKey | []uint8
}

// VerifyKeySet is a set of public or secret keys. It is used by the parser to verify a token.
type VerifyKeySet struct {
	Keys []VerifyKey
}
```

I typically prefer the term "verify" over "validate" when it comes to confirming authenticity. The term "validate" (in this codebase and more broadly) often covers a more broad set of checks.

Reply:
"verify" makes sense, since `SigningMethod` implements both `Sign` and `Verify` and this is actually the key used for `Verify`.

As for `VerifyKey` or `VerificationKey`: that's up to coin toss I guess ;)

Reply:
Pushed the renamed types in 1e49d85a02cca6c93007deceb16f240cfd5fd5a6

Given it's an interface, and the Go standard lib prefers nouns, the `VerificationKey` name is likely the most idiomatic. 

However, `VerifyKeySet` vs. `VerificationKeySet` is a coin toss. No strong opinion on this one, the former is shorter.

Reply:
I think having one "verify" and one "verification" is confusing, so I would probably go for "verification" on both then.

Reply:
Updated, now we have parity:

```go
// VerificationKey represents a public or secret key for verifying a token's signature.
type VerificationKey interface {
	crypto.PublicKey | []uint8
}

// VerificationKeySet is a set of public or secret keys. It is used by the parser to verify a token.
type VerificationKeySet struct {
	Keys []VerificationKey
}
```